### PR TITLE
Update base OS info #268

### LINKED ISSUE #268: Update base OS info
In multiple places within our docs we reference our prior CentOS base. As this is now approaching a legacy status due to our imminent Rockstor 4 Stable release, which is 'Built on openSUSE', it is proposed that we add to the CentOS references that from Rockstor 4 onwards the base/donor OS is openSUSE Leap. Not specifying the version here is also recommended to avoid overly rapid aging of the info.

As we were previously know as CentOS based it would be fitting to keep these references in place and, for the time being at least, augment this OS info as indicated above. That way those viewing the docs expecting a CentOS base can see the prior association which has, after all, gotten us this far.

### PR DESCRIPTION
**Pull request by phillxnet**
Fixes #268
@FroggyFlox Ready for review.
Essentially a long list of small changes so shouldn't be too bad. All links checked and all rendering looks to be OK.

### This pull request's proposal
Update base OS info re the Rocsktor 4 "Built on openSUSE" endeavour.

Includes:
- Rewrite of initial "Welcome to Rockstor" section.
- Add Rockstor 4 "Built on openSUSE" to FAQ section.
- Improve update instructions in install.rst. Reword and update to incorporate the newer dual system and link to the more recent update channels doc section.
- Specify that only Rockstor 3 installer has memtest86+ included.
- Update Quick Start section with Rockstor 4 installer link. Indicating that v3 is now the older more complicated install option. Add Arch references now v4 has ARM64 compatibility.
- Include ARM64 / Pi4 references in min spec section.
- Add v4 reference to install doc re hardware compatibility.
- Includes a reference to reported Pi4 / RPi 400's working.
- Update mdraid boot drive howto to indicate v3 only.
- Remove beta status for new installer.
- Re-order PBP memtest86+ section to favour Rocsktor 4.
- Move 'enabled repos' info from installer-howto to FAQ page. Repos are an advanced element and our installer howto should be as accessible as possible.
- Add Pi4 reference to FAQ re hardware.
- Enforce 79 char line length on modified files. Some readability/formatting constraints/exception allowed.

### Checklist
- [x] With the proposed changes no Sphinx errors or warnings are generated.
- [x] I have added my name to the AUTHORS file, if required (descending alphabetical order).

### CODE REVIEW

**Comment by phillxnet** [comment]
@FroggyFlox Just to note:
The pr is incomplete re default Leap 15.3 profile repos: What we actually have currently is:
```
installer:~ # zypper lr -u
Repository priorities in effect:                                                                                                       (See 'zypper lr -P' for details)
      97 (raised priority)  :  1 repository
      99 (default priority) :  4 repositories
     105 (lowered priority) :  1 repository

# | Alias                              | Name                                                                                        | Enabled | GPG Check | Refresh | URI
--+------------------------------------+---------------------------------------------------------------------------------------------+---------+-----------+---------+------------------------------------------------------------------------------------------------------
1 | Leap_15_3                          | Leap_15_3                                                                                   | Yes     | (r ) Yes  | Yes     | http://download.opensuse.org/distribution/leap/15.3/repo/oss/
2 | Leap_15_3_Updates                  | Leap_15_3_Updates                                                                           | Yes     | (r ) Yes  | Yes     | https://download.opensuse.org/update/leap/15.3/oss/
3 | home_rockstor                      | home_rockstor                                                                               | Yes     | (r ) Yes  | Yes     | https://download.opensuse.org/repositories/home:/rockstor/openSUSE_Leap_15.3/
4 | home_rockstor_branches_Base_System | home_rockstor_branches_Base_System                                                          | Yes     | (r ) Yes  | Yes     | https://download.opensuse.org/repositories/home:/rockstor:/branches:/Base:/System/openSUSE_Leap_15.3/
5 | repo-backports-debug-update        | Update repository with updates for openSUSE Leap debuginfo packages from openSUSE Backports | No      | ----      | ----    | http://download.opensuse.org/update/leap/15.3/backports_debug/
6 | repo-backports-update              | Update repository of openSUSE Backports                                                     | Yes     | (r ) Yes  | Yes     | http://download.opensuse.org/update/leap/15.3/backports/
7 | repo-sle-debug-update              | Update repository with debuginfo for updates from SUSE Linux Enterprise 15                  | No      | ----      | ----    | http://download.opensuse.org/debug/update/leap/15.3/sle/
8 | repo-sle-update                    | Update repository with updates from SUSE Linux Enterprise 15                                | Yes     | (r ) Yes  | Yes     | http://download.opensuse.org/update/leap/15.3/sle/
```
However given the enabled and otherwise "repo-backports-*" and "repo-sle-*" repos are implicitly inherited I'd like to have these added under an separate docs issue so we can address their non explicit inclusion. I.e. we don't specify these in our wiki-ng config so it would be nice to have an issue that reference their 'auto' inclusion as part, presumably, of the Leap/Jump/Leap 15.3 changes.

Contextual upstream reference: https://news.opensuse.org/2021/07/19/leap-gains-maintenance-update-improvements/

File: installation/install.rst [hunk 1]
```diff
@@ -35,8 +35,10 @@ Hardware recommendations
 -------------------------
 
 There is nothing about Rockstor that requires special hardware. It's Linux, and
-specifically a CentOS/Fedora fork, so it can be installed on a wide range of
-commodity hardware. See :ref:`minsysreqs` for basic requirements.
+specifically Rockstor 4 is "Built on openSUSE" while v3 used CentOS, so it can
+be installed on a wide range of commodity hardware. With 4 gaining ARM64
```
Comment (FroggyFlox) [inline installation/install.rst:39]:
```suggestion
be installed on a wide range of commodity hardware, with 4 gaining ARM64
```

File: installation/install.rst [hunk 2]
```diff
@@ -64,33 +66,59 @@ You can find user stories and example builds on our
 2. Yet another recommendation used by some community members is HP Proliant
    Microserver Gen8.
 
+3. Raspberry Pi4 and RPi 400 have also both been reported as working as
+   intended via our newer Rockstor 4 "Build on openSUSE" variant. N.B. RPi 400
```
Comment (FroggyFlox) [inline installation/install.rst:70]:
```suggestion
   intended via our newer Rockstor 4 "Built on openSUSE" variant. *N.B.* RPi 400
```

File: installation/install.rst [hunk 3]
```diff
@@ -64,33 +66,59 @@ You can find user stories and example builds on our
...
+
 Upgrading Rockstor
 ------------------
-Rockstor is under continuous development and we push tested updates in small
-batches. These updates are easy to install. While we roll-out major releases
-that require complete OS install, we try to make these releases as infrequent
-as possible. We strive to make most updates non-disruptive.
+Rockstor is under continuous development and we generally release updates in
+small batches. These updates are easy to install. But we do roll-out major
+releases that require a complete re-installs. Upgrading from Rockstor 3 (CentOS
```
Comment (FroggyFlox) [inline installation/install.rst:77]:
```suggestion
releases that require a complete re-install. Upgrading from Rockstor 3 (CentOS
```

File: installation/install.rst [hunk 4]
```diff
@@ -64,33 +66,59 @@ You can find user stories and example builds on our
...
+based) to Rockstor 4 "Built on openSUSE" is one such update.
+But such updates are very rare.
+
+Non re-install Rockstor updates can be installed in two ways :
+
+1. Install updates from the Web-UI (recommended):
+On the Rockstor Web-UI, far top-right, you will see an upward facing arrow if
```
Comment (FroggyFlox) [inline installation/install.rst:84]: This seems like a good spot for a screenshot focused on the "upward-facing arrow".

File: installation/install.rst [hunk 5]
```diff
@@ -64,33 +66,59 @@ You can find user stories and example builds on our
...
+any Rockstor package updates are available. All upstream packages except the
+'rockstor' package can similarly be installed by clicking on a flashing
+wifi-like icon. Again this is not shown if no updates are available. But it
```
Comment (FroggyFlox) [inline installation/install.rst:87]: This also seems like a good spot for a screenshot of the "wifi-like" icon.
Additional minor edit suggestion:
```suggestion
wifi-like icon. Again, this is not shown if no update is available, but it
```

File: installation/install.rst [hunk 6]
```diff
@@ -64,33 +66,59 @@ You can find user stories and example builds on our
...
+does show, as per the up-arrow, if there is a rockstor package update
+available.
+But only the up-arrow will actually update the main rockstor package.
+These disparate but related mechanisms allows folks to choose to only update
```
Comment (FroggyFlox) [inline installation/install.rst:91]:
```suggestion
These disparate but related mechanisms allow users to choose to only update what they want:
```

File: installation/install.rst [hunk 7]
```diff
@@ -64,33 +66,59 @@ You can find user stories and example builds on our
...
+all packages bar the rockstor package (wifi-like icon).
```
Comment (FroggyFlox) [inline installation/install.rst:92]: Suggestion below (pardon the potential lack of formatting for the moment):
```suggestion
- Wifi-like icon: all packages bar the rockstor package.
- Up arrow: only the rockstor package.
```

File: installation/install.rst [hunk 8]
```diff
@@ -64,33 +66,59 @@ You can find user stories and example builds on our
...
 
-Rockstor updates can be installed in two ways :
+See the following section for details on upgrading the Rockstor package plus
+all pending upstream updates :ref:`update_channels`.
 
-1. Install updates from the Web-UI : On the Rockstor Web-UI, on the top-right
-side of the navigation bar, you will see an upward facing arrow. If you click
-on that arrow, and if there is an update available, you will see *Software
-Update* highlighted, with details on the screen.
+2. Alternatively, for advanced users only, one can updated from the Command
```
Comment (FroggyFlox) [inline installation/install.rst:97]:
```suggestion
2. Alternatively, for advanced users only, one can update from the Command
```

File: installation/pre-install-howto.rst [hunk 9]
```diff
@@ -16,11 +16,37 @@ reasonably with the readily available tools.
...
+built on. Memtest86+ is licensed under the GPL.
+
+Memtest86+ Cautionary Note
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Memtest86+ can place a very intense load on your system, especially if run in
+the new Multithreaded mode (version 5.01 and onwards via F2 on initial start)
+but a sufficiently cooled system should be able to execute this test
+indefinitely; however, if your system has cooling issues then it may lockup
+or even sustain damage. Please take care to monitor your systems temperate
```
Comment (FroggyFlox) [inline installation/pre-install-howto.rst:30]:
```suggestion
or even sustain damage. Please take care to monitor your system's temperatures
```

File: installation/pre-install-howto.rst [hunk 10]
```diff
@@ -16,11 +16,37 @@ reasonably with the readily available tools.
...
+during this test. Version 5.01 and on have a built in CPU temperature monitor.
```
Comment (FroggyFlox) [inline installation/pre-install-howto.rst:31]:
```suggestion
during this test. Version 5.01 and on have a built-in CPU temperature monitor.
```

File: installation/pre-install-howto.rst [hunk 11]
```diff
@@ -16,11 +16,37 @@ reasonably with the readily available tools.
...
+If you find that your version of the Rockstor installer doesn't include the 5
+.01 or newer version of memtest86+, or any version at all (Rockstor 4) then
+download a `Pre-Compiled Bootable ISO <https://www.memtest.org/#downiso>`_
+from the projects official page and write it to a USB key in exactly the same
```
Comment (FroggyFlox) [inline installation/pre-install-howto.rst:35]:
```suggestion
from the project's official page and write it to a USB key in exactly the same
```

File: installation/quickstart.rst [hunk 12]
```diff
@@ -17,18 +17,20 @@ Rockstor is a complete Linux distribution (iso) to be installed directly on
...
   recommended).
 * Ethernet interface (with internet access for updates).
 * All drives must have unique serial numbers (real drives do); not all VM [*]_
   systems default to this.
 * A UPS (recommended) that is supported by `NUT <https://networkupstools.org/>`_.
-* USB port and 1GB+ USB key for the installation media; or DVD drive and a
-  blank DVD.
+* USB port and 1GB+ USB key for the x86_64 installation media; or a DVD drive.
+  Rasbperry Pi 4 / RPi 400 installs are via a self expanding system disk image,
+  and so do not required separate install media.
```
Comment (FroggyFlox) [inline installation/quickstart.rst:32]:
```suggestion
  and so do not require a separate install media.
```

File: installation/quickstart.rst [hunk 13]
```diff
@@ -75,21 +77,69 @@ USB install disk.
 Command line options
 ~~~~~~~~~~~~~~~~~~~~
 
-To create a USB install disk on Linux or Mac one can use the dd command. For
-example if your USB device is /dev/sdc then from within the directory
-containing your downloaded Rockstor iso file the command would be:-
+To create a USB install disk on Linux or Mac one can use the dd command.
+For example if your USB device is /dev/sdc then from within the directory
+containing your downloaded or self built
```
Comment (FroggyFlox) [inline installation/quickstart.rst:82]:
```suggestion
containing your downloaded or self-built
```

File: installation/quickstart.rst [hunk 14]
```diff
@@ -75,21 +77,69 @@ USB install disk.
...
-    dd if=Rockstor-3.8-9.iso of=/dev/sdc
+If you **built your own Pi4 installer** via our
+`rockstor-installer <https://github.com/rockstor/rockstor-installer>`_
+instructions then you can forgo the initial xzcat extraction step and use
+(single line command):
 
-Note that the .iso file name will vary depending on the Rockstor version.
+    dd bs=4M if=Rockstor-Leap15.2-RaspberryPi4.aarch64-4.0.7-0.raw of=/dev/sdc iflag=fullblock conv=notrunc status=progress
+
+For the more technically interested we create our xz download image files from
```
Comment (FroggyFlox) [inline installation/quickstart.rst:111]:
```suggestion
For the more technically interested, we create our xz download image files from
```

File: installation/quickstart.rst [hunk 15]
```diff
@@ -75,21 +77,69 @@ USB install disk.
...
+the raw installer created by the `kiwi-ng <https://github.com/OSInside/kiwi>`_
+system that our
+`rockstor-installer <https://github.com/rockstor/rockstor-installer>`_ is a
+configuration for via "xz --threads=4 --memlimit-compress=80% Rockstor-...raw"
+to enable multi-threaded decompress.
+
+**The ARM64EFI generic images**, when downloaded, are available in both the
+raw.zx file format, like the Pi4 images, and in pre-sized (16GB) qcow2 formats.
+For the raw.zx downloaded files the single line command is identical to the Pi4
```
Comment (FroggyFlox) [inline installation/quickstart.rst:120]:
```suggestion
For the raw.zx downloaded files, the single line command is identical to the Pi4
```

File: introduction/concepts.rst [hunk 16]
```diff
@@ -72,5 +72,6 @@ infrastructure.
 Rock-ons
 --------
 
-This is our made up term to reference the built in `docker <https://www.docker.com/>`_
-based plugin system.  See our :ref:`rockons_intro` section for a more detailed explanation.
+This is our made up term to reference the built in
```
Comment (FroggyFlox) [inline introduction/concepts.rst:75]:
```suggestion
This is our made-up term to reference the built-in
```

**Review by FroggyFlox — CHANGES_REQUESTED** [review]
Hi @phillxnet ,

Thanks for all the improvements throughout the docs; there was a lot to cover here!

I only have minor suggestions related to style or typos. I haven't had the time to test build or see the result in a local server but I thought I would share my suggestions now anyway.

Thanks again!

File: faq.rst [hunk 17]
```diff
@@ -70,36 +70,43 @@ Do you have any hardware recommendations?
 
 For home or small business use, we've seen Rockstor install flawlessly on
 HP, DELL and Supermicro servers and desktops. The developers of Rockstor
-use HP Micro servers for individual use.
+have use HP Micro servers for individual use.
```
[the remark this answers is not in the document]
Comment (phillxnet) [reply faq.rst:73]: Yes, I meant "have used"  and have now reworded in a pending commit.
Thanks.

File: faq.rst [hunk 18]
```diff
@@ -70,36 +70,43 @@ Do you have any hardware recommendations?
...
 How frequent are the software updates?
 --------------------------------------
 
-Rockstor development happens at a reasonably fast pace. We create test releases
-almost every other day. Once we are satisfied with a batch of changes, we
-release a stable update. This happens roughly once or twice a month.
+Rockstor development depends on the number of our contributors & Stable Channel
+subscribers. Testing channel releases are always more frequent.
+
+Once we are assured off all recent changes in the Testing Channel we release a
```
[the remark this answers is not in the document]
Comment (phillxnet) [reply faq.rst:98]: Are yes, moving along :).
Thanks.

File: installation/install.rst (hunk 4 above)
Comment (phillxnet) [reply installation/install.rst:84]: Agreed, I'll add via another commit.

File: installation/install.rst (hunk 5 above)
Comment (phillxnet) [reply installation/install.rst:87]: OK. Again I'll add via a commit as merging here might be tricky given I've now also added a pic in here via other changes. I may be in for a merge conflict with all these in GitHub and outside of GitHub changes on the same file. Oh well.

File: installation/install.rst (hunk 7 above)
Comment (phillxnet) [reply installation/install.rst:92]: Skipping this change as I need to double check all this actually.

File: installation/install.rst (hunk 6 above)
Comment (phillxnet) [reply installation/install.rst:91]: OK, this was likely a little too colloquial. Cheers.

File: installation/install.rst (hunk 7 above)
Comment (FroggyFlox) [reply installation/install.rst:98]: Yeah, I hesitated on that one but that's how I think I remembered it and your edits comforted what I remembered... I should have been more careful and verified myself first.

**Comment by phillxnet** [comment]
@FroggyFlox Thanks for the review. Much appreciated. I've popped in almost everything and have opened the following issue to address the one thing I skipped in your review. Just wanted to double check our function and the entire section needs a clean anyway:

https://github.com/rockstor/rockstor-doc/issues/299

Added some of you suggestion via non GitHub commits to aid in the merging as the picture addition involved some text re-jigging. Also pushed a commit to remove the now unused picture.

I think we both tried to get too much into a single pr here and I've unfortunately not managed to keep a clean git history either. Oh well, not so much of an issue here in docs so I'm pressing on and merging so we can get it published. I've re-tested post all changes and all affected/altered pages look to render OK.

**Pull request merged**